src/psyclone/psyir/nodes/dynamic_omp_task_directive.py

```diff
@@ -0,0 +1,1998 @@
+                    binop = BinaryOperation.create(
+                        node.operator, real_ref.copy(), step
+                    )
+                    # If modulo is non-zero then we need a second binop
+                    # dependency to handle the module, so we have two
+                    # dependency clauses, one which handles each of the
+                    # step-sized array chunks that overlaps with this access.
+                    if modulo != 0:
+                        if step2 is not None:
+                            binop2 = BinaryOperation.create(
+                                node.operator, real_ref.copy(), step2
+                            )
+                        else:
+                            binop2 = real_ref.copy()
+                else:
+                    # We have Literal OP Ref
+                    binop = BinaryOperation.create(
+                        node.operator, step, real_ref.copy()
+                    )
+                    # If modulo is non-zero then we need a second binop
+                    # dependency to handle the module, so we have two
+                    # dependency clauses, one which handles each of the
+                    # step-sized array chunks that overlaps with this access.
+                    if modulo != 0:
+                        if step2 is not None:
+                            binop2 = BinaryOperation.create(
+                                node.operator, step2, real_ref.copy()
+                            )
+                        else:
+                            binop2 = real_ref.copy()
+                # Add this to the list of indexes
+                if binop2 is not None:
+                    index_list.append([binop, binop2])
+                else:
+                    index_list.append(binop)
+
+        # Proxy use case handled.
+        # If the variable is private:
+        elif index_private:
+            # If the variable is in our private list
+            if ref in private_list:
+                # If its a child loop variable
+                if ref.symbol in child_loop_vars:
+                    # Return a full range (:)
+                    dim = len(index_list)
+                    one = Literal(str(dim + 1), INTEGER_TYPE)
+                    # Find the arrayref
+                    array_access_member = ref.ancestor(ArrayMember)
+                    # If there is an ArrayMember ancestor of this index, then
+                    # this is a StructureReference. For that case we need
+                    # to find the StructureReference parent, and recreate
+                    # the full a%b%c(...) style structure.
+                    if array_access_member is not None:
+                        start = ref.ancestor(StructureReference)
+                        sub_ref = StructureReference(start.symbol)
+                        sub_ref.addchild(start.member.copy())
+                        array_member = sub_ref.walk(Member)[-1]
```

Comment:
I thought it would - If you have `a%b%c%d(i)` then start is the `a` StructureReference, and `start.member.copy()` should give you "b%c%d(1)` as a copy, as I believe that it does a deep copy of the node's children.

Am I missing something?

Reply:
Oh yes, I don't know what I was thinking there. Ignore me :-)